pytorch_lightning/trainer/logging.py

```diff
@@ -196,8 +196,8 @@ def reduce_distributed_output(self, output, num_gpus):
             elif isinstance(output[k], torch.Tensor) and output[k].dim() == 0:
                 pass
 
-            # reduce only metrics that have the same number of gpus
-            elif output[k].size(0) == num_gpus:
-                reduced = torch.mean(output[k])
-                output[k] = reduced
+            # do not reduce metrics that have batch size > num gpus
```

Comment:
batch size has nothing to do with dp.... why is this fix even needed? 

size(0) should be the number of GPUs in DP... NOT batch size

Reply:
simple example: batch_size = 2, num_gpus = 3. Lightning will forward the batch with only 2 gpus, so the number of outputs is 2, so size(0) = 2. Therefore Lightning will not reduce the output and we get a problem later when the progress bar metrics call .item on that tensor. 

This is my explanation after hours and hours of debugging. If it's not that, I will give up.